### PR TITLE
HGCal Validation Plots

### PR DESCRIPTION
**Pull request by jingyucms**
#### PR description:

This PR adds TOT plots for HGCal Digi validation, and it also includes some minor changes of the histogram names/scales in the HGCal validation codes. 

#### PR validation:

The changes have been run in workflow 27434.0, 27634.0, and 27634.99 and produce plots for example in https://indico.cern.ch/event/801871/contributions/3337371/attachments/1804302/2944004/PreMixingValidation_update.pdf page 4. 

#### if this PR is a backport please specify the original PR:
No need to backport at this moment. 

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc [hunk 1]
```diff
@@ -215,8 +222,11 @@ void HGCalDigiValidation::analyze(const edm::Event& iEvent,
 	const HGCSample&  hgcSample = it.sample(SampleIndx_);
 	uint16_t   gain      = hgcSample.toa();
 	uint16_t   adc       = hgcSample.data();
-	double     charge    = adc*gain;
-	digiValidation(detId, geom0, layer, adc, charge);
+	//double     charge    = adc*gain;
```
Comment (kpedro88) [inline Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: what motivates this change?

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc [hunk 2]
```diff
@@ -336,9 +354,17 @@ void HGCalDigiValidation::fillOccupancyMap(std::map<int, int>& OccupancyMap,
 
 void HGCalDigiValidation::fillDigiInfo(digiInfo& hinfo) {
   int ilayer = hinfo.layer;
-  charge_.at(ilayer)->Fill(hinfo.charge);
-  DigiOccupancy_XY_.at(ilayer)->Fill(hinfo.x, hinfo.y);
-  ADC_.at(ilayer)->Fill(hinfo.adc);
+  //charge_.at(ilayer)->Fill(hinfo.charge);
```
Comment (kpedro88) [inline Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: why is this histogram removed?

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc (hunk 1 above)
Comment (jingyucms) [reply Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: To my understanding, I don't think energy weighted TOA is used in the simulation/reconstruction. So I simply change it to TOA. Let me know if I misunderstood.

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc (hunk 2 above)
Comment (jingyucms) [reply Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: The original histogram is filling the TOA information. So I just renamed the histogram from "charge_" to "TOA_".

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc (hunk 1 above)
Comment (kpedro88) [reply Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: @amartelli @Saptaparna can you take a look and confirm if this makes sense?

**Comment by andrius-k** [comment]
please test

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc (hunk 1 above)
Comment (amartelli) [reply Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: Hi @kpedro88 @jingyucms, 
I agree that adc*toa is not interesting and not used. 
Instead having a plot dedicated to the ToA is  very useful. 

thanks for this

**Comment by Saptaparna** [comment]
Yes, this is something I am working on.

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc (hunk 2 above)
Comment (kpedro88) [reply Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: Okay, then please delete all commented-out lines of code in the file.

Comment (jingyucms) [reply Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: Done removing commented-out lines.

**Comment by kpedro88** [comment]
please test

**Comment by andrius-k** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**